gtsam/hybrid/HybridFactorGraph.h

```diff
@@ -24,176 +25,259 @@
 
 namespace gtsam {
 
-class HybridFactorGraph {
+// Forward declarations
+class GaussianMixture;
+class Dummy;
+class HybridFactorGraph;
+class HybridEliminationTree;
+class Ordering;
+
+/** Main elimination function for HybridFactorGraph */
+GTSAM_EXPORT std::pair<GaussianMixture::shared_ptr, boost::shared_ptr<Factor>>
+EliminateHybrid(const HybridFactorGraph& factors, const Ordering& keys);
+
+template <>
+struct EliminationTraits<HybridFactorGraph> {
+  typedef Factor FactorType;
+  typedef HybridFactorGraph FactorGraphType;
+  typedef GaussianMixture ConditionalType;
+  typedef HybridBayesNet BayesNetType;
+  typedef HybridEliminationTree EliminationTreeType;
+  typedef HybridBayesNet BayesTreeType;
+  typedef HybridEliminationTree JunctionTreeType;
```

Comment:
`using` instead of `typedef`s?

Reply:
Want to keep this similar to GTSAM for now.